torch/_torch_docs.py

```diff
@@ -4709,6 +4709,45 @@ def merge_dicts(*dicts):
     tensor([  2.,   4.,   8.,  16.])
 """.format(**common_args))
 
+add_docstr(torch.nanprod,
+           r"""
+nanprod(input, dtype=None) -> Tensor
+
+Returns the product of all elements excluding ``NaN`` values in the
+:attr:`input` tensor.
+
+Args:
+    {input}
+    {dtype}
+
+Example::
+
+    >>> a = torch.tensor([float('nan'), 2, 3, 6, float('nan')])
+    >>> out = torch.nanprod(a)
+    >>> out
+    tensor(36.)
+
+.. function:: nanprod(input, dim, keepdim=False, dtype=None) -> Tensor
+
+Returns the product of each row of the :attr:`input` tensor in the given
```

Comment:
Returns the product of each row of the input tensor in the given dimension dim, treating "NaN" values as ones.

If keepdim is True, the output tensor is of the same size as input except in the dimension dim where it is of size 1. Otherwise, dim is squeezed (see torch.squeeze()), resulting in the output tensor having 1 fewer dimension than input.

Reply:
Done, thanks!